src/directives/admonitions.ts

```diff
@@ -22,10 +22,12 @@ class BaseAdmonition extends Directive {
     const newTokens: Token[] = []
 
     // we create an overall container, then individual containers for the title and body
+    const kind = this.title.toLowerCase().replace(/ /g, "")
 
     const adToken = this.createToken("admonition_open", "aside", 1, {
       map: data.map,
-      block: true
+      block: true,
+      meta: { kind, ...data.options }
```

Comment:
Almost all other directives don't "subclass", so this is a bit of a special case for admonitions. For all others, we are looking to the `_open` tag.

From the AST perspective, I think that note is a kind of admonition. Not a new AST type, this makes all sorts of things simpler downstream (i.e. it is more of a class, than having different structure or special parsing rules.)

We are doing a similar thing for mapping various types of math to one ast class, which is standard already in MDAST.

Reply:
> Almost all other directives don't "subclass",

of the ones currently in this package, but that won't be the case for extensible directives, and it also means that we basically have to "special case" every directive, if you want to traverse them in the AST, rather than just visiting a single `directive` type

Say you add a directive that creates the following tokens:

```xml
<token1>
<token2>
<token3>
```

where does the directive metadata go? Or are you enforcing that every directive can only generate a single "top level token (which is not the case for docutils)

Reply:
As mentioned on slack, the `kind`is a bit of a red herring here, thats ok (although I would probably just have another explicit class variable for that, separate from `title`),
its more the duplication of storing the directive options

Reply:
Added kind to the admonitions, and centralized the logic!